### PR TITLE
Pdct 495 add families to collections - update

### PR DESCRIPTION
**Pull request by katybaulch**
# Description

- Add families to collections by changing the collections on a family
- Added validation of import IDs of added/modified collections
- Added validation of collection organisations against current user organisation
- Validation of family organisation against current user organisation already implemented for update

[Link to Linear ticket](https://linear.app/climate-policy-radar/issue/PDCT-495/add-families-to-collections-implement)

## Type of change

Please select the option(s) below that are most relevant:

- [ ] Bug fix
- [x] New feature
- [ ] Breaking change

## How Has This Been Tested?

Updated unit and integration tests.

## Reviewer Checklist

- [ ] The PR represents a single feature (small driveby fixes are also ok)
- [ ] The PR includes tests that are sufficient for the level of risk
- [ ] The code is sufficiently commented, particularly in hard-to-understand areas
- [ ] Any required documentation updates have been made
- [ ] Any TODOs added are captured in future tickets
- [ ] No FIXMEs remain

### CODE REVIEW

File: app/service/family.py [hunk 1]
```diff
@@ -104,11 +111,32 @@ def update(
     if family is None:
         raise ValidationError(f"Could not find family {import_id}")
 
+    # Get the organisation from the user's email
+    user_org_id = app_user.get_organisation(db, user_email)
+
     # Validate metadata
```
Comment (diversemix) [inline app/service/family.py:117]: Think this comment should be nearer line 123

File: app/service/family.py [hunk 2]
```diff
@@ -104,11 +111,32 @@ def update(
...
     metadata.validate(db, org_id, family_dto.metadata)
 
-    if family_repo.update(db, import_id, family_dto, geo_id):
+    # Validate that the collections we want to update are from the same organisation as
+    # the current user and are in a valid format.
+    all_cols_to_modify = set(family.collections).union(set(family_dto.collections))
+
+    id.validate_multiple_ids(all_cols_to_modify)
+
+    collections_in_usr_org = [
```
Comment (diversemix) [inline app/service/family.py:131]: should this be perhaps named `collections_not_in_user_org`

Comment (katybaulch) [reply app/service/family.py:131]: Good spot! Forgot to change after changing from not all to any!

File: integration_tests/setup_db.py [hunk 3]
```diff
@@ -76,17 +76,33 @@
         "documents": ["D.0.0.1", "D.0.0.2"],
         "collections": ["C.0.0.2"],
     },
+    # {
```
Comment (diversemix) [inline integration_tests/setup_db.py:79]: Opps is this for later?

**Review by diversemix — APPROVED** [review]
Nice - very thorough ! :elephant:

File: integration_tests/setup_db.py (hunk 3 above)
Comment (katybaulch) [reply integration_tests/setup_db.py]: Yep! Good spot - forgot to take out.

**Pull request merged**